src/helpers.php

```diff
@@ -246,3 +246,37 @@ function is_countable($obj)
         return is_array($obj) || $obj instanceof Countable;
     }
 }
+
+if (! function_exists('oldValueDefaultOrFallback')) {
+    /**
+     * This function allows us to setup a default value in case there is an old value, but is forcelly null
+     * by Laravel middleware ConvertEmptyStringsToNull.
+     *
+     * @param  string  $field_name
+     * @param  mixed  $fallback
+     * @return mixed
+     */
+    function oldValueDefaultOrFallback($field, $fallback, $field_name = null, $default = null, $value = null)
+    {
+        $field_name = $field_name ?? $field['name'] ?? null;
+        $default = $default ?? $field['default'] ?? null;
+        $value = $value ?? $field['value'] ?? null;
+
+        // When you submit your field empty, internaly laravel will convert it to «null».
+        // Using old('field') with coalescing operator (??) we can't match it even if we have it in the old() session (as null).
+        // This checks that the value is null, but exists in the session old array, so it was submited empty|null, but exists!
+        if (! empty(session()->getOldInput())) {
+            if (is_null(old(square_brackets_to_dots($field_name)))) {
+                return $fallback;
+            }
+
+            return old(square_brackets_to_dots($field_name));
+        }
+
+        if (array_key_exists('value', $field)) {
+            return $value;
+        }
```

Comment:
@pxpm I don't understand - what is $value ? No, not in a philosophical sense, but here in this function 😅
How is it different from `$field['value']`? When would anybody pass that last parameter? We already have `fallback` as a last resort...

Reply:
So from what I understand, it's there for `date_range`. Because that has not one... but two inputs. So there you want to specify a custom field name, a custom default and a custom value.

So that's the answer to my question. And that's another reason for us NOT to make this helper depend on the field, at all.

Reply:
@tabacitu I guess an image is worth a thousand words:

![image](https://user-images.githubusercontent.com/7188159/147269015-23d9886b-fc72-41b6-8fbb-2776ef6ab32d.png)

BC on date_range ?

Reply:
Let's try non-BC. Please explain what's missing from `old_input_value()` in my other comment here. That function also avoid the complexity of this having $field inside the helper function, so it should work well inside `date_range` too.